## The purpose of this github repository is to demonstrate analysis and visualization of data from the 2016 Nature paper, *Proteogenomics connects somatic mutations to signalling in breast cancer* by Mertins et al.

**This analysis and visualization project was performed as a final project in the course 605.622 Data Visualization, part of the computer science program at Johns Hopkins Engineering for Professionals program (Fall 2023).**

The data to be visualized are a proteomic dataset of iTRAQ ratios of 80 breast cancer and 3 healthy tissue mass spectrometry analysis. The data are visualized in a heatmap, volcano plot (fold change, breast cancer vs. healthy proteome expression), and principle component analysis. In particular the principle component analysis is used to visualize data post machine learning clustering (k-means).

To begin, a new workbook containing the data we wish to work with was created from sheet Global-Proteome-G3 in "CPTAC_BC_SupplementaryTable03.xlsx". This new workbook was named "83_bc_prot.xlsx" and is the basis for the heatmap and volcano plots. In order to create a heatmap of all the genes, a custom R script "clean_data.R" is utilized. This script imports and cleans the data, including reproducing the author's original methods of discarding genes with indeterminate measurements and whose standard deviations were less than 1.5. 

Running "clean_data.R" on "83_bc_prot.xlsx" generates "77_cancer_proteomes_CPTAC_itraq_standev.csv", which is subsequently utilized by a custom Python script "heatmap_less_NaN_standev.py" to generate a heatmap. This script makes use of the *plotly* and *pandas* packages for data parsing and visualization. This python script also renames the samples to breast cancer and healthy before plotting. Running "heatmap_less_NaN_standev.py" on "77_cancer_proteomes_CPTAC_itraq_standev.csv" generates a heatmap:


![heatmap](https://github.com/slang314/proteogenomics-reproduction/assets/155842228/549c56bc-2f9b-4721-9b9d-ba335d34206f)


The legend is shown at right where yellow and green indicate high log_2 iTRAQ ratios and dark or purple colors indicate low or negative ratios. Given that the genes were simply assigned an integer, this quick visualization is meant to look for any interesting grouping patterns. The heatmap could be improved by adding the gene name on the left instead of an integer and adding the PAM50 subtype to the breast cancer names on the y-axis.

Next, Excel was utilized to concatonate columns and add statistics between the two groups, such as foldchange, p-value (via student's t-test), and negative log_2 of the fold change. This allows us to create a [Volcano Plot](https://en.wikipedia.org/wiki/Volcano_plot_(statistics)), a common method of "quickly identifying changes in large data sets composed of replicate data." The methods for this step were borrowed from [_Guide for protein fold change and p-value calculation for non-experts in proteomics_](https://pubs.rsc.org/en/content/articlelanding/2020/mo/d0mo00087f) by Aguilan et al. The resultant file was named "modified_names_droppednas.csv". Running "volcano.py" on "modified_names_droppednas.csv" file produces a volcano plot:

![newplot(4)](https://github.com/slang314/proteogenomics-reproduction/assets/155842228/486ceccb-d4c7-4525-8a4d-278048cc0770)

The python script was written to add statistics in the title. Comparing cancer samples vs. normal we see that in cancer cells 1,646 genes were found to be underexpressed as per a protein measurement and 6,348 are found to be overexpressed. This is consistent with what we know about [cancer global overexpression or hyperexpression of genes](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9683723/) as well as [cancer global DNA hypomethylation](https://www.nature.com/articles/nature18003) enabling higher expression levels as it is typically promoter methylation which silences a given gene, among other mechanisms. 

Like the heatmap, the plot is interactive, and statistics and gene names appear upon hovering:

![image](https://github.com/slang314/proteogenomics-reproduction/assets/155842228/e7d73170-557a-4f8f-ab5c-dc0ac627e2da)


In this manner, one can explore the data at their leisure. Additionally, a subregion of the plot my be selected for closer inspection of genes of interest. The original input can be mined via other scripts for the greatest foldchange and -log_2(p-value) statistics.
Next, a k-means clustering on the input samples was performed via the Python machine learning library _scikit-learn_ (sklearn). The basis for this plot was an excel file to which I added information regarding the sample, the original cluster, and the PAM50 subtype. This was done given that in particular I was interested in reproducing similar clusters to the original analysis. A csv file "kmeans_final.csv" was thus generated by manually reading and parsing information from the original supplmentary tables. This file was used as input into "kmeansv3.py", a custom script utilizing pandas, sklearn, and plotly to perform mathematics and data visualization. Running this script generates the following plot:

![newplot(3)](https://github.com/slang314/proteogenomics-reproduction/assets/155842228/4a1f0c4f-38f8-4da8-990f-48829a99cbc8)

Unfortunately, the reproducibility of the clustering was highly sensitive to the initial random state set when calling the kmeans clustering function, and I am still working to understand precisely what is meant by this. My first notion is that the precise code, packages, and random state (if entered) utilized for the authors initial clustering needs to be more precisely understood, along with misunderstandings which may exist for me as a student of machine learning.


**Sources Cited:**

Mertins, P., Mani, D., Ruggles, K. et al. Proteogenomics connects somatic mutations to signalling in breast cancer. Nature 534, 55â€“62 (2016). https://doi.org/10.1038/nature18003
